internal/service/organizations/organizational_unit_descendant_accounts.go

```diff
@@ -0,0 +1,189 @@
+package organizations
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/organizations"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func DataSourceOrganizationalUnitDescendantAccounts() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceOrganizationalUnitDescendantAccountsRead,
+
+		Schema: map[string]*schema.Schema{
+			"parent_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"accounts": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"arn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"email": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"joined_method": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"joined_timestamp": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// Todo: Get all child OUs
+func getDescendantOrganizationalUnitsIDs(parent_id string, meta interface{}) ([]string, error) {
+	conn := meta.(*conns.AWSClient).OrganizationsConn
+
+	params := &organizations.ListOrganizationalUnitsForParentInput{
+		ParentId: aws.String(parent_id),
+	}
+
+	// Descendants will hold IDs of all generations of OUs under parent_id.
+	var descendants []string
+	// Children will hold immediate child OUs of parent_id.
+	var children []*organizations.OrganizationalUnit
+	// ChildOUIds will hold the OU IDs of child ous.
+	var childOUIds []string
+	// result will hold any descendants of immediate child OUs.
+	var result []string
+
+	// Get all immediate children OUs.
+	err := conn.ListOrganizationalUnitsForParentPages(params,
+		func(page *organizations.ListOrganizationalUnitsForParentOutput, lastPage bool) bool {
+			children = append(children, page.OrganizationalUnits...)
+
+			return !lastPage
+		})
+
+	if err != nil {
+		return nil, fmt.Errorf("error listing Organizations Organization Units for parent (%s): %w", parent_id, err)
+	}
+
+	if childOUIds = getIDsFromOUs(children); childOUIds != nil {
+		for _, id := range childOUIds {
+			// Append this child ou Id and all it's descendants Ids.
+			descendants = append(descendants, id)
+
+			result, err = getDescendantOrganizationalUnitsIDs(id, meta)
+			if err != nil {
+				return descendants, err
+			} else if descendants != nil {
+				descendants = append(descendants, result...)
+			} else {
+				return nil, nil
+			}
+		}
+
+		return descendants, nil
+	}
+
+	//Base Case. ParentId has no children.
+	return nil, nil
```

Comment:
Since you could have some results at this point, could you return those and then ditch the entire `internal/service/organizations/organizational_unit_child_accounts.go` file?

Reply:
Right, so if both data sources return the same thing in that case, then we don't need the other one. Right? If you can stick all the logic into a single data source, then consumers don't need to know their organization in order to call the right one.

Reply:
I had it in my head that there would be a valid use case where I had an OU that contained both child OUs and immediate child accounts, and I wanted just those immediate child accounts and not the other "generations" of children. If there was only one data source you would only be able to get all generations of child accounts. 

I think that's still a valid use case, but i guess it's debatable whether it's a common use case.